packages/astro/src/runtime/server/index.ts

```diff
@@ -367,13 +387,17 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 		if (metadata.hydrate === 'only') {
 			html = await renderSlot(result, slots?.fallback);
 		} else {
-			({ html } = await renderer.ssr.renderToStaticMarkup.call(
-				{ result },
-				Component,
-				props,
-				children,
-				metadata
-			));
+			try {
+				({ html } = await renderer.ssr.renderToStaticMarkup.call(
+					{ result },
+					Component,
+					props,
+					children,
+					metadata
+				));
+			} catch (e) {
+				return e as string;
```

Comment:
Hm, I'm not sure why we're returning the error as a string here. Couldn't this render errors to the DOM instead of hoisting to the error overlay?

Reply:
Oops, nice catch. This should throw.

Reply:
Fixed!